src/core/mac/mac.hpp

```diff
@@ -101,6 +101,139 @@ enum
     kIndirectFrameMacTxAttempts = OPENTHREAD_CONFIG_MAX_TX_ATTEMPTS_INDIRECT_PER_POLL,
 };
 
+/**
+ * This class defines a channel mask.
+ *
+ * It is a wrapper class around a `uint32_t` bit vector representing a set of channels.
+ *
+ */
+class ChannelMask
+{
+public:
+    /**
+     * This constructor initializes a `ChannelMask` instance.
+     *
+     */
+    ChannelMask(void)
+        : mMask(0)
+    {
+    }
+
+    /**
+     * This constructor initializes a `ChannelMask` instance with a given mask.
+     *
+     * @param[in]  aMask   A channel mask (as a `uint32_t bit-vector mask)
+     *
+     */
+    ChannelMask(uint32_t aMask)
+        : mMask(aMask)
+    {
+    }
+
+    /**
+     * This method clears the channel mask.
+     *
+     */
+    void Clear(void) { mMask = 0; }
+
+    /**
+     * This method gets the channel mask (as a `uint32_t` bit-vector mask).
+     *
+     * @returns The channel mask.
+     *
+     */
+    uint32_t GetMask(void) const { return mMask; }
+
+    /**
+     * This method sets the channel mask.
+     *
+     * @param[in]  aMask   A channel mask (as a `uint32_t bit-vector mask)
```

Comment:
Mention endian-ness (i.e. which bit corresponds to channel 0).

typo: missing back tick